File: facade/golang/coldfood.go
package main

type ColdFood struct {
	name string
}

func NewColdFood(name string) *ColdFood {
	return &ColdFood{name: name}
}

func (c *ColdFood) WashAndRinse() string {
	return "Washing and rinsing cold food: " + c.name
}

func (c *ColdFood) Wrap() string {
	return "Wrapping cold food: " + c.name
}

func (c *ColdFood) Freeze() string {
	return "Freezing cold food: " + c.name
}

func (c *ColdFood) Prepare() string {
	return "Preparing cold food: " + c.name
}
func (c *ColdFood) Serve() string {
	return "Serving cold food: " + c.name
}

File: facade/rust/facade/src/coldfood.rs
pub struct ColdFood {}

impl ColdFood {
    pub fn washAndRinse(&self) -> &str {
        "ColdFood Washed And Rinsed"
    }

    pub fn wrap(&self) -> &str {
        "ColdFood Wrapped"
    }

    pub fn freeze(&self) -> &str {
        "ColdFood frozen"
    }
}